pkg/review/commit_review_status.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package review contains code to get review status information
+// for a GitHub commit.
+
+package review
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/shurcooL/githubv4"
+)
+
+func GetPullRequests(client *githubv4.Client, githubOrg, repository, commitSha string) ([]PullRequest, error) {
+	var query struct {
+		Repository struct {
+			Object struct {
+				Commit struct {
+					AssociatedPullRequest struct {
+						Nodes    []PullRequest
+						PageInfo struct {
+							EndCursor       githubv4.String
+							HasNextPage     githubv4.Boolean
+							HasPreviousPage githubv4.Boolean
+							StartCursor     githubv4.String
+						}
+						TotalCount githubv4.Int
+					} `graphql:"associatedPullRequests(first: 100, after: $pageCursor)"`
+				} `graphql:"... on Commit"`
+			} `graphql:"object(oid: $commitSha)"`
+		} `graphql:"repository(owner: $githubOrg, name: $repository)"`
+	}
+	getPage := func(cursor githubv4.String) error {
+		err := client.Query(context.Background(), &query, map[string]interface{}{
+			"githubOrg":  githubv4.String(githubOrg),
+			"repository": githubv4.String(repository),
+			"commitSha":  githubv4.GitObjectID(commitSha),
+			"pageCursor": cursor,
+		})
+		if err != nil {
+			return fmt.Errorf("GitHub GraphgQL call failed: %w", err)
+		}
+		return nil
+	}
```

Comment:
Unless there is a reason I am missing, I feel like this is overkill, see example in docs: https://github.com/shurcooL/githubv4?tab=readme-ov-file#pagination

Reply:
Getting the first page separately lets us use the `TotalCount` to preallocate the slice capacity. This prevents us from needing to resize the slice as we append to it.

Reply:
Although I don't have a problem with getting the first page outside the loop, I don't think resizing slice should be a main consideration here.

Reply:
Well i think getting the first page separately is also cleaner than using an infinite loop with a `break` as the example does.